From a431db3cd256312a57ce1d5dec9d70dd1e65d86f Mon Sep 17 00:00:00 2001
From: Aman Pandae <AmanPandae.Mothukuri@windriver.com>
Date: Tue, 30 Dec 2025 05:24:52 -0500
Subject: [PATCH] Add support to 'app.starlingx.io/component' to be overwritten
 by user override, with possible values being 'platform' and 'application'.
 With 'platform' being the default value. This change will also restart the
 pods if the label in namespace change.

Signed-off-by: Aman Pandae <AmanPandae.Mothukuri@windriver.com>
---
 templates/daemonset.yaml | 2 ++
 values.yaml              | 3 +++
 2 files changed, 5 insertions(+)

diff --git a/templates/daemonset.yaml b/templates/daemonset.yaml
index 3b49c0e..fc6cac5 100644
--- a/templates/daemonset.yaml
+++ b/templates/daemonset.yaml
@@ -46,6 +46,7 @@ spec:
         istio.io/dataplane-mode: none
         app.kubernetes.io/name: {{ template "name" . }}
         {{- include "istio.labels" . | nindent 8 }}
+        app.starlingx.io/component: {{ ternary "application" "platform" (eq .Values.component "application") }}
         {{ with .Values.podLabels -}}{{ toYaml . | nindent 8}}{{ end }}
       annotations:
         sidecar.istio.io/inject: "false"
@@ -60,6 +61,7 @@ spec:
         # securityContext which is otherwise preferred.
         container.apparmor.security.beta.kubernetes.io/install-cni: unconfined
         # Custom annotations
+        configchecksum: {{ .Values.component | quote | sha256sum }}
         {{- if .Values.podAnnotations }}
 {{ toYaml .Values.podAnnotations | indent 8 }}
         {{- end }}
diff --git a/values.yaml b/values.yaml
index d7caeee..259223d 100644
--- a/values.yaml
+++ b/values.yaml
@@ -43,6 +43,9 @@ _internal_defaults_do_not_set:
   # Additional labels to apply on the pod level
   podLabels: {}

+  # App label
+  component: platform
+
   # Deploy the config files as plugin chain (value "true") or as standalone files in the conf dir (value "false")?
   # Some k8s flavors (e.g. OpenShift) do not support the chain approach, set to false if this is the case
   chained: true
--
2.34.1

